third_party/netcdf-c.BUILD

```diff
@@ -0,0 +1,31 @@
+'''
+netcdf-c@4.9.0
+'''
+
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "allsrcs",
+    srcs = glob(["**"]),
+)
+
+cmake(
+    name = "netcdf-c",
+    cache_entries = {
+        "BUILD_SHARED_LIBS": "ON",
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_INSTALL_LIBDIR": "lib",
+        "ENABLE_DAP": "OFF",
+        "ENABLE_EXAMPLES": "OFF",
+        "ENABLE_PNETCDF": "OFF",
+        "ENABLE_TESTS": "OFF",
+        "USE_SZIP": "OFF",
+    },
+    generate_args = ["-GNinja"],
```

Comment:
Let's stick with make for all of our `cmake` dependencies.

Ninja is great for incremental rebuilds, but for a fresh build there's not much advantage over pure make. Since these are third party dependencies and won't change often, I think we can live without. One less dependency for our build to have to concern itself with.